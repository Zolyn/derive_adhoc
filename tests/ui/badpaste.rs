use derive_adhoc::{derive_adhoc, Adhoc};

#[derive(Adhoc)]
struct DataType {
    foo: u8,
    bar: Vec<String>,
}

derive_adhoc! {
    DataType:

    ${for fields {
        ${paste $ttype _ $ftype}
    }}
}

derive_adhoc! {
    DataType:

    struct ${paste $tname _ 42};

    struct ${paste $tname ${tmeta(something)}};
}

fn main() {}
