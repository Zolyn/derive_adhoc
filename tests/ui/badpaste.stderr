error: multiple nontrivial entries in ${paste ...}
  --> ui/badpaste.rs:13:18
   |
13 |         ${paste $ttype _ $ftype}
   |                  ^^^^^

error: multiple nontrivial entries in ${paste ...}
  --> ui/badpaste.rs:13:27
   |
13 |         ${paste $ttype _ $ftype}
   |                           ^^^^^

error: not allowed in within ${paste ...} (or case_changing)
  --> ui/badpaste.rs:20:29
   |
20 |     struct ${paste $tname _ 42};
   |                             ^^
